boxsdk/object/file.py

```diff
@@ -433,3 +471,199 @@ def get_tasks(self, fields=None):
             fields=fields,
             return_full_pages=False,
         )
+
+    @api_call
+    def get_previous_versions(self, limit=None, offset=None, fields=None):
+        """
+        Get previous versions of the file.
+
+        :param limit:
+            The maximum number of items to return per page. If not specified, then will use the server-side default.
+        :type limit:
+            `int` or None
+        :param offset:
+            The index at which to start returning items.
+        :type offset:
+            `int`
+        :param fields:
+            List of fields to request.
+        :type fields:
+            `Iterable` of `unicode`
+        :returns:
+            An iterator of the previous versions of the file.
+        :rtype:
+            :class:`BoxObjectCollection`
+        """
+        return LimitOffsetBasedObjectCollection(
+            session=self.session,
+            url=self.get_url('versions'),
+            limit=limit,
+            fields=fields,
+            offset=offset,
+            return_full_pages=False,
+        )
+
+    @api_call
+    def promote_version(self, file_version):
+        """
+        Promote a file version to become the current version of this file.  This will create a new file version
+        identical to the previous version as the new current version.
+
+        :param file_version:
+            The file version to promote.
+        :type file_version:
+            :class:`FileVersion`
+        :returns:
+            The new file version created as the current.
+        :rtype:
+            :class:`FileVersion`
+        """
+        url = self.get_url('versions', 'current')
+        body = {
+            'type': 'file_version',
+            'id': file_version.object_id,
+        }
+        response = self._session.post(url, data=json.dumps(body)).json()
+        return self.translator.translate(
+            session=self._session,
+            response_object=response,
+        )
+
+    @api_call
+    def delete_version(self, file_version, etag=None):
+        """
+        Delete a specific version of a file.
+
+        :param file_version:
+            The file version to delete.
+        :type file_version:
+            :class:`FileVersion`
+        :param etag:
+            If specified, instruct the Box API to update the item only if the current version's etag matches.
+        :type etag:
+            `unicode` or None
+        :returns:
+            Whether the operation succeeded.
+        :rtype:
+            `boolean`
+        """
+        url = self.get_url('versions', file_version.object_id)
+        headers = {'If-Match': etag} if etag is not None else None
+        response = self._session.delete(url, expect_json_response=False, headers=headers)
```

Comment:
This should just be `file_version.delete()`

Do we really need this method?

Reply:
This is the same as in the Email Alias deletion; the URL requires both the file ID and the version ID.